aliquotmaf/annotators/gnomad.py

```diff
@@ -40,24 +40,35 @@ def empty_gnomad_record():
 
 
 class GnomAD(Annotator):
-    def __init__(self, scheme, refpath, refpattern):
+    def __init__(self, scheme, ref_prefix):
         super().__init__(name="GnomAD", scheme=scheme)
-        self.path = refpath
-        self.file_template = refpattern
+        self.file_template = ref_prefix + "{}.feather"
         self.chrom = None
         self.df = None
 
     @classmethod
-    def setup(cls, scheme, refpath, refpattern):
+    def setup(cls, scheme, ref_prefix):
         """
         Prepare annotation sources
 
-        refpath should point to directory containing annotation files
-        refpattern should be a template with which to create the file names
-            by substituting chromosome names in for {}
-            it should look like:  reference.{}.feather
+        ref_prefix: prefix used to locate reference files it will be prepended
+            to strings in the form of 'chr1.feather' in order to construct
+            the path to the reference files
         """
-        curr = cls(scheme, refpath, refpattern)
+        curr = cls(scheme, ref_prefix)
+        missing = [
+            chr
+            for chr in CHROM_LIST
+            if not os.path.exists(curr.file_template.format(chr))
+        ]
+        if missing is []:
```

Comment:
Doesn't it become much harder to understand that way though? The computational logic is opposed to the way it reads in English.

Reply:
There might be a logic bug here actually:
The warning should print the list of chromosomes without corresponding files, but it would only get there if `missing == []`.

So it should really be `if missing:`, right?